pkg/util/limiter/utilization_test.go

```diff
@@ -35,23 +35,44 @@ func TestUtilizationBasedLimiter(t *testing.T) {
 	}
 
 	t.Run("CPU based limiting should be enabled if set to a value greater than 0", func(t *testing.T) {
-		lim, _, reg := setup(t, 0.11, gigabyte)
+		lim, scanner, reg := setup(t, 0.11, gigabyte, true)
+
+		var prevTotalTime float64
+		samples := make([]float64, 60)
 
 		// Warmup the CPU utilization.
 		for i := 0; i < int(resourceUtilizationSlidingWindow.Seconds()); i++ {
 			lim.compute(nowFn)
 			tim = tim.Add(time.Second)
+			for i := 1; i < len(samples); i++ {
+				samples[i-1] = samples[i]
+			}
+			s := scanner.totalTime - prevTotalTime
+			samples[len(samples)-1] = s
+			prevTotalTime = scanner.totalTime
```

Comment:
In this unit test you're basically copy-pasting the code from the implementation. Not sure it increase much the confidence. Instead of here, I would rather test it in the "CPU sensitivity" test where you you can just generate the expected last 60 samples using the CPU generation functions used in the test itself.

Reply:
Thanks, will try.

Reply:
I arrived at making a dedicated test case for checking `lim.cpuSamples` in an obvious way (as static as possible), it was the simplest solution for me at least.